near/kycdao-ntnft/src/lib.rs

```diff
@@ -116,32 +160,89 @@ impl KycdaoNTNFT {
-            token_statuses: UnorderedMap::new(StorageKey::TokenStatuses),
+            authorized_statuses: UnorderedMap::new(StorageKey::AuthorizedStatusesV0_4_0),
+            token_statuses: UnorderedMap::new(StorageKey::TokenStatusesV0_4_0),
+            subscription_cost_per_year: 5 * u32::pow(10, SUBSCRIPTION_COST_DECIMALS as u32),
+            authorized_seconds_to_pay: UnorderedMap::new(StorageKey::AuthorizedSecondsToPay),
+            authorized_tiers: UnorderedMap::new(StorageKey::AuthorizedTiers),
+            token_tiers: UnorderedMap::new(StorageKey::TokenTiers),
+            native_usd_price_feed,
         }
     }
 
     #[private]
     #[init(ignore_state)]
     pub fn migrate() -> Self {
+        log!("Starting migration to v0.4.0...");
+
         let old_state: OldKycdaoNTNFT = env::state_read().expect("failed");
+
+        log!("Old state read successfully");
+
+        let mut new_authorized_statuses = UnorderedMap::new(StorageKey::AuthorizedStatusesV0_4_0);
+        for (key, old_status) in old_state.authorized_statuses.iter() {
+            let new_status = Status {
+                verified: !old_status.is_revoked,
+                expiry: old_status.expiry,
+            };
+            new_authorized_statuses.insert(&key, &new_status);
+        }
+
+        log!("Migrated authorized_statuses");
+
+        let mut new_token_statuses = UnorderedMap::new(StorageKey::TokenStatusesV0_4_0);
+        for (key, old_status) in old_state.token_statuses.iter() {
+            //log!("Migrating old token status: {:?}: {:?}", key, old_status);
+            let new_status = Status {
+                verified: !old_status.is_revoked,
+                expiry: old_status.expiry,
+            };
+            new_token_statuses.insert(&key, &new_status);
+        }
+
+        log!("Migrated token_statuses");
+
+        let price_feed_addr = AccountId::from_str("price-oracle.testnet").expect("accountID should be valid");
+        let native_usd_price_feed = PriceFeed::new(price_feed_addr);
+
+        log!("Defined price feed");
+
         Self {
             tokens: old_state.tokens,
             metadata: old_state.metadata,
             next_token_id: old_state.next_token_id,
             mint_authorizer: old_state.mint_authorizer,
             authorized_token_metadata: old_state.authorized_token_metadata,
-            authorized_statuses: UnorderedMap::new(StorageKey::AuthorizedStatuses),
-            token_statuses: UnorderedMap::new(StorageKey::TokenStatuses),
+            authorized_statuses: new_authorized_statuses,
+            token_statuses: new_token_statuses,
+            subscription_cost_per_year: 5 * u32::pow(10, SUBSCRIPTION_COST_DECIMALS as u32),
+            authorized_seconds_to_pay: UnorderedMap::new(StorageKey::AuthorizedSecondsToPay),
+            authorized_tiers: UnorderedMap::new(StorageKey::AuthorizedTiers),
+            token_tiers: UnorderedMap::new(StorageKey::TokenTiers),
```

Comment:
There's no tier migration? Looking at the rest of the code now, I assume it's a matter of 'lets just return the default if it doesn't exist'. I think I'd prefer it to be a little more explicit...

Reply:
I realise there's now default handling for the tokens which don't have a tier set now. And I realise this is a small thing, but - I still feel like the default handling should be the 'oh no, I don't know what happened but lets just return the default', not a 'lets just handle this well known case with defaults'. As in, it should be a fallback when things unexpectedly go wrong. In this case, we have the chance to migrate all the existing tokens to the default tier here. Which means we don't have to rely on the fallback later on. It just feels like a loose end to have tokens around that don't have a tier explicitly set.